### ISSUE TITLE
iOS: Support ITF Tags (not only ITF-14)

### ISSUE THREAD

**Issue by rdesimone** [phonegap/phonegap-plugin-barcodescanner#455]
### Expected Behaviour
It should  be possible to scan an ITF tag and a valid result will be returned. (The ITF tags we are using have 10 or less digits)

### Actual Behaviour
The ITF tag is not recognised (other tags like e.g. EAN are recognised). The Android version of the plugin is able to scan the mentioned ITF tags.

### Reproduce Scenario (including but not limited to)
With the simplest Cordova app scan the attached tag.

#### Platform and Version
cordova-ios@4.3.1
iOS 10.3.1

#### Cordova CLI version and cordova platform version
6.5.0

#### Plugin version
6.0.5
6.0.6

## Additional Remark
For our case we solved the problem by using this old fork: https://github.com/chen6516/cordova-zxing-portrait (it was mentioned in this issue https://github.com/phonegap/phonegap-plugin-barcodescanner/issues/354#issuecomment-260945979 )

So at some point it seems that ITF was supported. Maybe it also a problem of the ZXing version. Anyhow IMO it is an issue because these are popular tags.

![screen shot 2017-04-21 at 15 13 35](https://cloud.githubusercontent.com/assets/5596378/25279671/82818dda-26a7-11e7-99b4-9ffaac711872.png)

**Comment by EddyVerbruggen** [comment]
As mentioned in Slack: on iOS there's [only platform support for ITF 14](https://github.com/phonegap/phonegap-plugin-barcodescanner/blob/b5a135c6788e1a9ecb9f06a087931701f0b7a321/src/ios/CDVBarcodeScanner.mm#L533) (not 8,10,12,14 like on Android).

The fork you found uses a version that has support for other types but the iOS impl changed because of memory leak issues. See #312.

**Comment by rdesimone** [comment]
@EddyVerbruggen I would recommend to leave the issue open. There are two reasons: 

1. it is is not confirmed that 14 digit tags are working (at least this I understood in Slack - nobody tested).
2. the lack of support for ITF tags with less digits is a serious problem

With or without memory leak, in the past less digits were supported.

At a certain point we can rename the issue when there is a confirmation that 14 digits are supported.
Think about it!

**Comment by EddyVerbruggen** [comment]
I don't mind leaving this open. Please update when you know more.

**Comment by rdesimone** [comment]
Thanks, yes anyway I have to investigate the issue more carefully.

**Comment by rdesimone** [comment]
@EddyVerbruggen I run the app in the debugger and checked memory consumption. IMO there is no memory leak with this forked plugin (see above). When opening the scanner memory increases about 20MB. When closing the scanner memory it is the same as before. I made about 15 scans in a row. 

The forked code looks quite different. I assume the plugin evolved a lot during the last two years. 

Does the current version still use ZXing? The MetadataObjectType look the same as the iOS API.
``` 
[output setMetadataObjectTypes:@[AVMetadataObjectTypeQRCode,
                                     AVMetadataObjectTypeAztecCode,
                                     AVMetadataObjectTypeDataMatrixCode,
                                     AVMetadataObjectTypeUPCECode,
                                     AVMetadataObjectTypeEAN8Code,
                                     AVMetadataObjectTypeEAN13Code,
                                     AVMetadataObjectTypeCode128Code,
                                     AVMetadataObjectTypeCode93Code,
                                     AVMetadataObjectTypeCode39Code,
                                     AVMetadataObjectTypeITF14Code,
                                     AVMetadataObjectTypePDF417Code]];
```
It would be interesting to understand why this forked plugin is able to scan the mentioned tags.

This is for me a sign that something fundamentally changed: 

```
        DecodeHints decodeHints;
        decodeHints.addFormat(BarcodeFormat_QR_CODE);
        decodeHints.addFormat(BarcodeFormat_DATA_MATRIX);
        decodeHints.addFormat(BarcodeFormat_UPC_E);
        decodeHints.addFormat(BarcodeFormat_UPC_A);
        decodeHints.addFormat(BarcodeFormat_EAN_8);
        decodeHints.addFormat(BarcodeFormat_EAN_13);
        decodeHints.addFormat(BarcodeFormat_CODE_128);
        decodeHints.addFormat(BarcodeFormat_CODE_39);
        decodeHints.addFormat(BarcodeFormat_ITF);
```

Thanks!

**Comment by wittmamz** [comment]
Hello,

I have the same issue with ITF Barcodes containing 8 digits. Are there any news regarding this problem? @rdesimone does your solution work without crashes in the long-term?

Thanks!

**Comment by rdesimone** [comment]
Yes, the solution works in production in an international airport app - no crashes! But just to let you know: we will produce different kind of baggage tags in the future which are also supported by the iOS API. If you can control the tag creation, it is maybe better to do this. Otherwise you have to fork a version which uses the ZXing library - like I did. I don't know if I am correct but the current plugin version supports only the tags which are supported by the iOS API.

**Comment by wittmamz** [comment]
Thank you @rdesimone, the plugin works. But I did notice the memory leak issue mentioned above. My app's memory usage increased by about 100MB when I spent a while scanning (I don't know how many scans I did, maybe around 50).

**Comment by rdesimone** [comment]
@wittmamz thanks for your feedback! As written above, I could not see an increase - maybe I have to check again. Anyhow, in case there is a memory leak, for our app it is not a super tragedy (even though ugly) because we have very little scans per app use. In case you find an alternative solution, please let me know!

**Comment by john-doherty** [comment]
Hello. Any progress on this?

**Comment by jcesarmobile** [comment]
There is a PR, but I requested some changes. You can see it linked right over your comment

**Comment by jcesarmobile** [comment]
Closed by #636
